File: src/main/java/fi/eonwe/wikilinks/fatpages/WikiPageData.java
package fi.eonwe.wikilinks.fatpages;

import java.util.List;

/**
 *
 */
public record WikiPageData(String title, int id, List<PagePointer> links) implements WikiPage {

    @Override
    public boolean isRedirect() {
        return false;
    }
}

File: src/main/kotlin/fi/eonwe/wikilinks/fatpages/WikiPageData.kt
package fi.eonwe.wikilinks.fatpages

/**
 * 
 */
data class WikiPageData(val title: String, val id: Int, val links: List<PagePointer>) : WikiPage {

    override val isRedirect: Boolean = false

    override fun id(): Int = id

    override fun title(): String = title
}
